### DISCUSSION TITLE
Unknown profile is shown as having commited my code

### DISCUSSION THREAD

**Discussion by petarchou** [community/community#56837] [Repositories] [answered]
### Select Topic Area

Bug

### Body

Bug Summary: Commits are being assigned incorrectly to accounts that have nothing to do with the repository.

In [this](https://github.com/petarchou/Ethernaut) repository of mine, some of the commits are shown to be made by a [random github account](https://github.com/h2zh).
I am 100% certain I made all the commits and that there are no collaborators in the project.
I even contacted the person and, obviously, he has no idea what's going on either.

**Comment by airtower-luna** [comment, marked as answer]
GitHub assigns commits to users based on the *author email* recorded in the commit. The commits in question [seem to be using `placeholder@gmail.com`](https://github.com/petarchou/Ethernaut/commit/c75befd678151ef70effefe6148632e7974e6b01.patch), which seems odd to say the least. :sweat_smile:

Make sure your [user name and email are set correctly](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#_your_identity), for GitHub to recognize your commits `user.email` must be an address connected to your account.

If you want to fix the existing commits, you'll need to [modify the repository history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History). Note that this will change *all* commit IDs after the first changed commit. This means anyone who already has a copy of your repository and tries to fetch updates will have to manually accept the modified history. :warning:

**Reply by petarchou** [reply]
Well, that expains it I guess. 
I thought it would use the account I've  logged in with on my machine, not just the git email address I've set.

Doesn't that mean that I can change my email and make it seem as if someone else uploaded some malicious code, lets say?

**Reply by PurpleMonkey729** [reply]
> Well, that expains it I guess. I thought it would use the account I've logged in with on my machine, not just the git email address I've set.
> 
> Doesn't that mean that I can change my email and make it seem as if someone else uploaded some malicious code, lets say?

Mr Petarchou, Nice to meet you.
What's the meaning of "Doesn't that mean that I can change my email and make it seem as if someone else uploaded some malicious code, lets say?"?
What is the purpose of that malicious action?
I think there is no reason that anyone should do like that thing.

**Reply by airtower-luna** [reply]
> Doesn't that mean that I can change my email and make it seem as if someone else uploaded some malicious code, lets say?

The same way you could add a "by Someone Else" phrase to your project readme, yes. Which could be anything from an obnoxious prank to a serious crime (impersonation can have a lot of consequences).

In working with Git it's common to move someone else's commits around. Imagine working with someone on a feature, and when you're done the branch with the new code has commits from both of you. Then one of you want to get that branch merged, maybe push it somewhere else first. Obviously that'll involve pushing (or making a PR for) code from multiple authors. And then you have the workflow of sending patches by email (like when working on the Linux kernel), where the recipient (if they accept the patches) creates new commits containing the patches, with the author(s) listed correctly.

**Reply by PurpleMonkey729** [reply]
I see. Thanks for you kind description.
